tests/saw/spec/handshake/handshake_io_lowlevel.saw

```diff
@@ -64,13 +64,12 @@ let config_cca_type config = (crucible_field config "client_cert_auth_type");
 let ocsp_status_size cert_and_key =
     crucible_field (crucible_field (cert_and_key) "ocsp_status") "size";
 
-//conn->config->use_tickets is the field we care about, but it is
-// in a bitfield. For now, we only care about that, so we point
-// at the entire bitfield and say it must all be initialized to
-// a value of 0 for the proof to go through. This is a stronger
-// statement than we stricly need to make, and will limit
-// compositional proofs in the future, but we can make more
-// precise statements at that time
+// conn->config->use_tickets and conn->config->quic_enabled are the fields we
+// care about, but they are in a bitfield. We currently handle this by
+// initializing the entire bitfield with a symbolic value and imposing a
+// precondition that bit 3 (i.e., use_tickets) must be equal to 0.
+// Bit 9 (i.e., quic_enabled), on the other hand, is allowed to be either
+// 0 or 1.
```

Comment:
We've had a real problem on the s2n_connection structure with SAW failing when we add new bitflags to a bitfield. Is this going to trigger the same issue for s2n_config?

Reply:
The SAW code here was already accessing `s2n_config`'s bitfield by way of `use_tickets`, so in some sense, this PR isn't changing the status quo—we just access another bitflag within the same bitfield (`quic_enabled`) now. 

That being said, I definitely agree that having to interact with bitfields in such an indirect fashion is cumbersome, and it's something that we're keen to fix (time permitting, of course). One possible way to fix it is laid out in GaloisInc/crucible#366, although we've also been discussing an alternative fix that wouldn't require such a drastic rewrite of SAW's internals.

Reply:
Do you know why we can change s2n_config's bitflags but changing s2n_connection's bitflags breaks the proofs?

Reply:
I'd have to see the full details of the failure to know for sure. My guess is that if you add another field to `s2n_connection`'s first bitfield, it will change the memory offset of `s2n_connection`'s second bitfield. Currently, SAW proofs access bitfields in structs by directly specifying a memory offset to load, so changes to the struct's memory layout could affect SAW proofs which offset into it.

It's also worth noting that `s2n_connection`'s first bitfield in particular uses eight fields, which is just enough to be combined into a single byte. My understanding of how bitfields in C work is that adding even one more field to that bitfield would result in the bitfield expanding in size to two bytes (and therefore shifting all subsequent fields down one byte in memory). `s2n_config`'s bitfield, on the other hand, currently has 11 fields, which means that any SAW proofs accessing that bitfield shouldn't have to change even if up to five more fields are added.

Reply:
Could you add a comment explaining the support for bit-fields could be improved in the future by using metadata from the LLVM bitcode?

Reply:
I've expanded the comments to mention GaloisInc/saw-script#1461, which describes a possible fix for this issue, including the details about LLVM bitcode.